src/main/java/uk/gov/pay/directdebit/common/validation/ApiValidation.java

```diff
@@ -19,18 +19,54 @@
     private static final Logger LOGGER = PayLoggerFactory.getLogger(ApiValidation.class);
 
     private String[] requiredFields;
+    private Map<String, Integer> minimumFieldsSize;
     private Map<String, Integer> maximumFieldsSize;
     private Map<String, Function<String, Boolean>> validators;
 
-    public ApiValidation(String[] requiredFields, Map<String, Integer> maximumFieldsSize, Map<String, Function<String, Boolean>> validators) {
+    public ApiValidation(String[] requiredFields,
+                         Map<String, Integer> minimumFieldsSize,
+                         Map<String, Integer> maximumFieldsSize,
+                         Map<String, Function<String, Boolean>> validators) {
         this.requiredFields = requiredFields;
+        this.minimumFieldsSize = minimumFieldsSize;
         this.maximumFieldsSize = maximumFieldsSize;
         this.validators = validators;
     }
 
-    private boolean isFieldSizeValid(Map<String, String> chargeRequest, String fieldName, int fieldSize) {
-        Optional<String> value = Optional.ofNullable(chargeRequest.get(fieldName));
-        return !value.isPresent() || value.get().length() <= fieldSize; //already checked that mandatory fields are already there
+    public void validate(String externalId, Map<String, String> request) {
+        List<String> missingFields = checkMissingFields(request, requiredFields);
```

Comment:
We can think about refactoring this logic (run a validation, check if empty) because this will keep growing if we add more validators. For now, I guess it's ok 👍